src/server/socksServer.ts

```diff
@@ -96,10 +96,15 @@ class SocksV5ServerParser {
     const socket = this._socket;
     let i = 0;
     const readByte = () => chunk[i++];
+    const closeSocketOnError = () => {
+      socket.end();
+      debugLogger.log('proxy', `Closed Socket due error: ${new Error().stack}`);
```

Comment:
I wonder whether we should also resolve or reject `this._ready`? It depends on how we call it.